stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/CSS.java

```diff
@@ -33,18 +49,287 @@ public void disable(JsonRpcPeer peer, JSONObject params) {
   }
 
   @ChromeDevtoolsMethod
-  public JsonRpcResult getSupportedCSSProperties(JsonRpcPeer peer, JSONObject params) {
-    GetSupportedCSSPropertiesResponse response = new GetSupportedCSSPropertiesResponse();
-    response.cssProperties = Collections.emptyList();
-    return response;
+  public JsonRpcResult getComputedStyleForNode(JsonRpcPeer peer, JSONObject params) {
+    final GetComputedStyleForNodeRequest request = mObjectMapper.convertValue(
+        params,
+        GetComputedStyleForNodeRequest.class);
+
+    final GetComputedStyleForNodeResult result = new GetComputedStyleForNodeResult();
+    result.computedStyle = new ArrayList<>();
+
+    mDocument.postAndWait(new Runnable() {
+      @Override
+      public void run() {
+        Object element = mDocument.getElementForNodeId(request.nodeId);
+
+        if (element == null) {
+          LogUtil.e("Tried to get the style of an element that does not exist, using nodeid=" +
+              request.nodeId);
+
+          return;
+        }
+
+        mDocument.getElementStyles(
+            element,
+            new StyleAccumulator() {
+              @Override
+              public void store(String name, String value, boolean isDefault) {
+                if (!isDefault) {
+                  CSSComputedStyleProperty property = new CSSComputedStyleProperty();
+                  property.name = name;
+                  property.value = value;
+
+                  result.computedStyle.add(property);
+                }
+              }
+            });
+      }
+    });
+
+    return result;
+  }
+
+  @ChromeDevtoolsMethod
+  public JsonRpcResult getMatchedStylesForNode(JsonRpcPeer peer, JSONObject params) {
+    final GetMatchedStylesForNodeRequest request = mObjectMapper.convertValue(
+        params,
+        GetMatchedStylesForNodeRequest.class);
+
+    final GetMatchedStylesForNodeResult result = new GetMatchedStylesForNodeResult();
+
+    final RuleMatch match = new RuleMatch();
+
+    result.matchedCSSRules = ListUtil.newImmutableList(match);
+
+    match.matchingSelectors = ListUtil.newImmutableList(0);
+
+    Selector selector = new Selector();
+    selector.value = "<this_element>";
+
+    CSSRule rule = new CSSRule();
+
+    rule.origin = Origin.Regular;
+    rule.selectorList = new SelectorList();
+
+    rule.selectorList.selectors = new ArrayList<>();
+    rule.selectorList.selectors.add(selector);
+
+    rule.style = new CSSStyle();
+    rule.style.cssProperties = new ArrayList<>();
+
+    match.rule = rule;
+
+    rule.style.shorthandEntries = new ArrayList<>();
```

Comment:
since we aren't putting anything in here we can use `Collections.emptyList()` (save 2 allocations)